baseapp_profiles/graphql/mutations/roles.py

```
@@ -57,15 +61,33 @@ def mutate_and_get_payload(cls, root, info, **input):
+                [
+                    ProfileUserRole(user_id=user_pk, profile_id=profile_pk, role=role_type)
+                    for user_pk in requested_user_pks
+                ]
+            )
+        except IntegrityError:
+            # Safety net for a race between the check above and the insert.
+            logger.exception("Failed to add members to profile %s", profile_pk)
+            raise GraphQLError(
+                str(_("One or more of the selected users are already members of this profile")),
+                extensions={"code": "already_member"},
+            )
```

Ruff (0.15.20)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)